File: ProjetoPadraoDotnetCore/Aplication/Models/Response/Template/TemplateResponse.cs
namespace Aplication.Models.Response.Template;

public class TemplateResponse
{ 
    public int? IdTemplate { get; set; }
    public string? Titulo { get; set; }
    public string? Descricao { get; set; }
    public string? Foto { get; set; }
    public string? EscalaTempo { get; set; }
    public int? Quantidade { get; set; }
    public string? DataCadastro { get; set; }
    public string? Categoria { get; set; }
    public int? IdUsuarioCadastro { get; set; }
    public List<AtividadeTemplateResponse>? LAtividade { get; set; }
}

public class AtividadeTemplateResponse
{
    public int? TempoPrevisto { get; set; }
    public string? Titulo { get; set; }
    public List<TarefaTemplateResponse> LTarefaTemplate { get; set; }
    public int? Posicao { get; set; }
}

public class TarefaTemplateResponse
{
    public string? Descricao { get; set; }
    public string? DescricaoTarefa { get; set; }
    public string? Prioridade { get; set; }
    public List<string?> LTagsTarefa { get; set; }
}

File: ProjetoPadraoAngular/src/app/objects/Template/TemplateResponse.ts
import { RetornoPadrao } from "../RetornoPadrao";

export interface TemplateResponse extends RetornoPadrao
{
    data: DataTemplateResponse
}

export interface DataTemplateResponse
{
    idTemplate: number | undefined,
    titulo: string;
    descricao: string;
    quantidade: number;
    categoria: string;
    escalaTempo: string,
    descricaoCategoriaNova: string | undefined,
    idUsuarioCadastro: number,
    lAtividade: Array<AtividadeTemplateResponse>,
    foto: string | undefined,
    isView: boolean 
}


export interface AtividadeTemplateResponse
{
    tempoPrevisto: number,
    idTemplate: number | undefined,
    titulo: string,
    posicao: number,
    lTarefaTemplate: Array<TarefaTemplateResponse>
}

export interface  TarefaTemplateResponse
{
    descricao: string;
    descricaoTarefa: string;
    prioridade: string,
    lTagsTarefa: Array<string>
}

